core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -276,7 +276,8 @@ class KafkaController(val config : KafkaConfig, zkClient: ZkClient, val brokerSt
                     case e : IllegalStateException => {
                       // Resign if the controller is in an illegal state
                       error("Forcing the controller to resign")
-                      controllerElector.resign()
+                      brokerRequestBatch.clear
+                      controllerElector.resign
```

Comment:
nit: both clear() and resign() have side effects, so we need the parentheses. 
Other than this, LGTM.